types/react-i18next/src/context.d.ts

```diff
@@ -24,3 +30,47 @@ export interface i18NextModule {
 }
 
 export const reactI18nextModule: i18NextModule;
+
+export interface ReactI18nContext {
+    i18n: i18n;
+    defaultNS?: string;
+    reportNS?: string;
+    lng: string;
+    t: TranslationFunction;
+}
+
+export const I18nContext: React.Context<ReactI18nContext|undefined>;
+
+// Diff / Omit taken from https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766
+type Omit<T, K extends keyof T> = Pick<T, ({ [P in keyof T]: P } & { [P in K]: never } & { [x: string]: never, [x: number]: never })[keyof T]>;
+
+export type ContextInferableComponentEnhancerWithProps =
+    <P extends { [key: string]: any }>(component: React.ComponentClass<P> | React.StatelessComponent<P>) =>
+        React.ComponentClass<Omit<P, keyof ReactI18nContext> & React.ClassAttributes<never>>;
```

Comment:
I've been adding `React.ClassAttributes<never>` to the HOC to help avoid accidentally saving a `ref` to the HOC instead of the actually wrapped component. The `getWrappedInstance()` API is gone from react-i18next.

Reply:
I also tried using `React.ClassAttributes<React.ComponentClass>` but that was not causing an expected type error on tests.